xbmc/music/MusicDatabase.cpp

```diff
@@ -9010,7 +9024,14 @@ bool CMusicDatabase::GetBoxsetDiscs(const std::string& strBaseDir, CFileItemList
       pItem->SetLabel2(m_pDS->fv("iDisc").get_asString()); // GUI show label2 for disc sort order??
       pItem->GetMusicInfoTag()->SetDiscNumber(discnum);    // Skins show discnumber for "albums"?? Confluence does.
       pItem->SetLabel(strDiscSubtitle);
-
+      if (iArtRowsFound > 1) // skip if only 1 row (no custom covers set)
+      {
+        for (unsigned int i=0 ; i < coverart.size(); i+=2)
+        {
+          if (coverart[i] == StringUtils::Format("cover%i", discnum))
+            pItem->SetArt("poster", coverart[i+1]); // set the individual cover art url
+        }
```

Comment:
Disc specific cover art is already automatically fetched from folder.jpg held local to the disc music files and held as album art of type "thumb1", "thumb2", "thumb3" etc.  not "poster".

Reply:
Estuary is looking for a "poster" value in some of it's views.  This is why art shows in 2 of the views (but not the disc subtitle, just the original album title), not at all in the widelist view (just text, shows the disc title as text) but not the last.  If I set the 'poster' value, it does.  At the time, this seemed like a better (easier ?) option than faffing around with the skin although that could well have been the wrong approach when I think about it now.  Still, it worked !!  I tried various different approaches to try to get the custom art into 'thumb' without much success.  Similarly, there doesn't appear to be a way (not that I can see anyway) to alter or overwrite an album title so that it shows in all the different views.  It's not a big thing, and I can navigate alright with the info thats there but it isn't 100% correct in some views and that niggles me.  Guess it might need either another info label, some skin changes or a bit of both.

Reply:
Estuary is simply wrong (it was done by people that never use music lib and didn't give an dam). They even opened a forum thread for user bug feedback during beta while privately in team saying they had no intention of looking at any of it. And got praised! I have never forgiven them for such attitudes, (most) users deserve better. [Friday night honesty!]

I understand your desire and happiness at getting something that worked, and I'm sure what you did had reasons given what you could see, but yes some of it needs to be done in the skin. Also Confluence is much easier to follow and mod if you want to mock up something for testing.

Reply:
I can switch to using Confluence on my Dev system.  My actual 'production' systems use a heavily customised version of Estuary done by a forum member.  This version was never released publicly, but has some major modifications to the PVR section.  As we can only watch TV via Kodi & tvheadend the pvr part needs to be user friendly and this members mod is awesome.  I also rolled out a build of my branch to a libreelec install on one of my Pi's and to my main instance so that I can use the discs feature.  Both of these are using this modded skin (which I further modded to work with my radio helper) so that's why I looked at Estuary.  That and the fact that if something gives unexpected results in a skin, the first question is always "What happens if you use Estuary ?"

There are people that never use the music library !!??  Clearly they are missing out then.  It does go some way though to explaining why most views are set up with video library variables and some don't even contain any references to the music library, despite the same view being used for both.

Reply:
I always suggest users try a simpler skin like "Estuary _or Confluence_" ;)
You were right to test with Estuary, but wrong (understandably) to assume what it did was correct. Skinning isn't my thing so to test the art stuff I somethimes ask a team member to make me a skin hack, or I use debug a lot. If I can see that the art is there in the item artmap then that is good enough for me!